### PR TITLE
Address CI warnings to not use nodefaults channel

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Remove use of nodefaults channel, use conda-remove-defaults: true instead

This change is prompted by CI warning:

```
'nodefaults' channel detected: will remove 'defaults' if added implicitly.
In the future, 'nodefaults' as a way of removing 'defaults' won't be
supported. Please set 'conda-remove-defaults' = 'true' to remove this warning.
```

image

---

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] Have you added documentation for your changes, if necessary?
- [ ] Have you added your changes to the changelog?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

**Review by ndgrigorian — DISMISSED** [review]
Warnings are gone, LGTM

**Comment by oleksandr-pavlyk** [comment]
Deprecation warnings are gone. Future warnings are still present, but the log does not show `defaults` channel being active even though it may be temporarily added to the list of channels by conda.

File: .github/workflows/conda-package.yml [hunk 1]
```diff
@@ -800,7 +802,8 @@ jobs:
         with:
           run-post: false
           channel-priority: "disabled"
-          channels: conda-forge,nodefaults
+          channels: conda-forge
```
Comment (antonwolfy) [inline .github/workflows/conda-package.yml:805]: `upload_windows` job uses `conda-incubator/setup-miniconda` step, but there is `channels` property missing. I think it needs to be aligned with other usages of `conda-incubator/setup-miniconda` step and so both `channels: conda-forge` and `conda-remove-defaults: true` properties to be defined there.

Comment (antonwolfy) [reply .github/workflows/conda-package.yml:805]: Otherwise the base environment is creating with packages from `pkgs/main` channel which is restricted.

**Review by antonwolfy — APPROVED** [review]
Thank you @oleksandr-pavlyk, LGTM!

**Comment by ndgrigorian** [comment]
Unfortunately, the deprecated warnings appear to be back, for the Windows step

**Review by ndgrigorian — APPROVED** [review]
I misread in my last comment, it was the future warnings. LGTM!

**Pull request merged**